resolwe_bio/processes/junctions/regtools.yml

```diff
@@ -0,0 +1,232 @@
+      label: Table of annotated novel splice junctions
+      type: basic:file
+    - name: splice_junctions
+      label: Table of annotated splice junctions
+      type: basic:file
+    - name: bed
+      label: Splice junctions in BED format
+      type: basic:file
+    - name: bigbed_igv_ucsc
+      label: Splice junctions in BigBed format
+      type: basic:file
+      required: false
+    - name: tbi_jbrowse
+      label: Bed tbi index for JBrowse
+      type: basic:file
+    - name: species
+      label: Species
+      type: basic:string
+    - name: build
+      label: Build
+      type: basic:string
+  run:
+    runtime: polyglot
+    language: bash
+    program: |
+
+      {% if not alignment_star and not alignment and not input_bed_junctions %}
+        re-warning "Please define STAR alignment or BAM file or Junctions in BED12 format."
+      {% endif %}
+
+      {% if (alignment_star and alignment)
+        or ( alignment_star and input_bed_junctions)
+        or (alignment and input_bed_junctions ) %}
+        re-error "Please define only STAR alignment or BAM file or Junctions in BED12 format."
+      {% endif %}
+
+      {% if not genome.species == annotation.species %}
+        re-error \
+          {{ ('Species of genome (%s) and annotation (%s) do not match. Please provide genome ' ~
+             'and annotation with the same species.')|format(genome.species, annotation.species) }}
+      {% endif %}
+
+      {% if not genome.build == annotation.build %}
+        re-error \
+          {{ ('Build of genome (%s) and annotation (%s) do not match. Please provide genome ' ~
+             'and annotation with the same build.')|format(genome.build, annotation.build) }}
+      {% endif %}
+
+      {% if alignment_star %}
+
+        {% if not alignment_star.species == genome.species %}
+          re-error \
+            {{ ('Species of STAR splice junctions file (%s) and genome (%s) do not match. ' ~
+              'Please provide STAR splice junctions file and genome with the same species.')
+              |format(alignment_star.species, genome.species) }}
+        {% endif %}
+
+        {% if not alignment_star.build == genome.build %}
+          re-error \
+            {{ ('Build of STAR splice junctions file (%s) and genome (%s) do not match. ' ~
+              'Please provide STAR splice junctions file and genome with the same build.')
+              |format(alignment_star.build, genome.build) }}
+        {% endif %}
+
+        if [ ! -s {{ alignment_star.sj.file }} ]; then
+          re-warning \
+            "STAR SJ.out.tab file has no entries. There will be no splice junctions detected."
+        fi
+
+        NAME=`basename {{ alignment_star.sj.file }} _SJ.out.tab`
+
+        # calculate BED12 fields from STAR SJ.out.tab file
+        awk \
+          {'if($4=="2") print \
+            ""$1"\t"$2-$9-1"\t"$3+$9"\tJUNC000"NR"\t"$7"\t-\t"$2-$9-1"\t"$3+$9"\t255,0,0\t2\t"$9","$9"\t0,"$3-$2+$9+1""; \
```

Comment:
This part still hurts me :-) What are your suggestions for improving code readability? I am thinking of making a Python script `star_sj_to_bed12.py`...